### DISCUSSION TITLE
how to use GoogleOAuthProvider on Authentication in flet

### DISCUSSION THREAD

**Discussion by bobwatcherx** [flet-dev/flet#1108] [Q&A]
### Question

i want to implement google oauth on my project flet .  and i follow this tutorial https://flet.dev/docs/guides/python/authentication/

And i get error message 
 error NameError: name 'GoogleOAuthProvider' is not defined

is there documentation to use google oauth in fleet

### Code sample

```python
self.provider = GoogleOAuthProvider(
        client_id="PRIVATE!!!!",
        client_secret="PRIVATE !!!!!",
        redirect_url="http://localhost:8550/api/oauth/redirect",
    )
self.page.on_login = self.on_login
def on_login(self,e):
		print("acces token : ",self.page.auth.token.access_token)
		print("acces id : ",self.page.user.id)
```

### Error message

```bash
Unhandled error processing page session : Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/flet/flet.py", line 302, in on_session_created
    session_handler(page)
  File "/home/minx/belajar/fletApp/fletemploye/main.py", line 683, in main
    mylogin = MyLogin()
  File "/home/minx/belajar/fletApp/fletemploye/main.py", line 12, in __init__
    self.provider = GoogleOAuthProvider(
NameError: name 'GoogleOAuthProvider' is not defined
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by thetrotfreak** [comment]
- Did you import the `GoogleOAuthProvider` class? If not, then do `from flet.auth.providers.google_oauth_provider import GoogleOAuthProvider`
- Also, you'd need to invoke `self.page.login(self.provider)` to start the authorization flow.

You shouldn't be getting a `NameError` if you've imported it properly and that it is available in that scope.
